packages/protocol-kit/src/Safe.ts

```diff
@@ -1007,6 +1018,197 @@ class Safe {
     )
     return txResponse
   }
+
+  /**
+   * Returns the Safe Transaction encoded
+   *
+   * @async
+   * @param {SafeTransaction} safeTransaction - The Safe transaction to be encoded.
+   * @returns {Promise<string>} The encoded transaction
+   *
+   */
+  async getEncodedTransaction(safeTransaction: SafeTransaction): Promise<string> {
+    const safeSingletonContract = await getSafeContract({
+      ethAdapter: this.#ethAdapter,
+      safeVersion: await this.getContractVersion()
+    })
+
+    const encodedTransaction: string = safeSingletonContract.encode('execTransaction', [
+      safeTransaction.data.to,
+      safeTransaction.data.value,
+      safeTransaction.data.data,
+      safeTransaction.data.operation,
+      safeTransaction.data.safeTxGas,
+      safeTransaction.data.baseGas,
+      safeTransaction.data.gasPrice,
+      safeTransaction.data.gasToken,
+      safeTransaction.data.refundReceiver,
+      safeTransaction.encodedSignatures()
+    ]) as string
+
+    return encodedTransaction
+  }
+
+  /**
+   * Wraps a Safe transaction into a Safe deployment batch.
+   *
+   * This function creates a transaction batch of 2 transactions, which includes the
+   * deployment of the Safe and the provided Safe transaction.
+   *
+   * @async
+   * @param {SafeTransaction} safeTransaction - The Safe transaction to be wrapped into the deployment batch.
+   * @param {TransactionOptions} [transactionOptions] - Optional. Options for the transaction, such as from, gas price, gas limit, etc.
+   * @param {string} [customSaltNonce] - Optional. a Custom salt nonce to be used for the deployment of the Safe. If not provided, a default value is used.
+   * @returns {Promise<Transaction>} A promise that resolves to a Transaction object representing the prepared batch of transactions.
+   * @throws Will throw an error if the safe is already deployed.
+   *
+   */
+  async wrapSafeTransactionIntoDeploymentBatch(
+    safeTransaction: SafeTransaction,
+    transactionOptions?: TransactionOptions,
+    customSaltNonce?: string
+  ): Promise<Transaction> {
+    const isSafeDeployed = await this.isSafeDeployed()
+
+    // if the safe is already deployed throws an error
+    if (isSafeDeployed) {
+      throw new Error('Safe already deployed')
+    }
```

Comment:
This is nice but is already checked in the place where this method is used. So always being checked twice

https://github.com/safe-global/safe-core-sdk/blob/9202697675e2b43b8b63724a7c5c1757af74cd19/packages/relay-kit/src/packs/gelato/GelatoRelayPack.ts#L236-L248

Reply:
It's justified that we need to check this condition twice in the code because each check serves a different purpose:

- The `wrapSafeTransactionIntoDeploymentBatch` acts as a "utility" function that may be used in other places in the future. This function requires a check to ensure that the Safe is not already deployed, in that case, it wouldn't be possible to wrap a Safe transaction into a deployment batch. If the Safe is already deployed, an error should be thrown to prevent a failed transaction.
- On the other hand, the `isSafeDeployed` check present in the `executeRelayTransaction` function serves a different purpose. It helps to construct the transaction object properly before to be sent to the Gelato relayer. If the Safe is already deployed, the `target` field value will be the Safe's address, but if the Safe is not yet deployed, the `target` field value will be the MultiSend contract address.